src/main/java/lotto/domain/ticket/LottoTicket.java

```diff
@@ -1,63 +1,41 @@
 package lotto.domain.ticket;
 
 import lotto.domain.number.LottoNumber;
+import lotto.domain.number.LottoNumbers;
+import lotto.domain.prize.Prize;
+import lotto.exception.ErrorMessage;
 
-import java.util.Arrays;
-import java.util.LinkedHashSet;
 import java.util.List;
-import java.util.Set;
-import java.util.stream.Collectors;
 
 public class LottoTicket {
 
-    private static final String LOTTO_NUMBER_DELIMITER = ", ";
-    public static final int LOTTO_NUMBER_SIZE = 6;
+    private final LottoNumbers lottoNumbers;
 
-    private Set<LottoNumber> numbers;
-
-    private LottoTicket() {
-    }
-
-    private LottoTicket(final List<LottoNumber> numbers) {
-        validateNumbers(numbers);
-        this.numbers = new LinkedHashSet<>(numbers);
-    }
-
-    public static LottoTicket of(final List<Integer> numbers) {
-        return new LottoTicket(numbers.stream()
-                .map(LottoNumber::of)
-                .collect(Collectors.toList()));
+    LottoTicket(final LottoNumbers lottoNumbers) {
+        validate(lottoNumbers);
+        this.lottoNumbers = lottoNumbers;
     }
 
-    public static LottoTicket of(final String numbers) {
-        return new LottoTicket(Arrays.stream(numbers.split(LOTTO_NUMBER_DELIMITER))
-                .map(Integer::parseInt)
-                .map(LottoNumber::of)
-                .collect(Collectors.toList()));
+    public static LottoTicket of(final LottoNumbers lottoNumbers) {
+        return new LottoTicket(lottoNumbers);
     }
 
-    private void validateNumbers(final List<LottoNumber> numbers) {
-        if ((numbers == null) || (numbers.size() != LOTTO_NUMBER_SIZE)) {
-            throw new IllegalArgumentException();
+    private void validate(final LottoNumbers lottoNumbers) {
+        if (lottoNumbers == null) {
+            throw new IllegalArgumentException(ErrorMessage.NULL_VALUE);
         }
     }
 
-    public int matchedCount(final WinningLottoTicket winningLottoTicket) {
-        return Math.toIntExact(numbers.stream()
-                .filter(winningLottoTicket::contains)
-                .count());
+    public Prize matchPrize(final WinningLottoTicket winningLottoTicket) {
+        int matchCount = lottoNumbers.matchCount(winningLottoTicket);
```

Comment:
```java
private final LottoNumbers lottoNumbers;
````
를
```java
protected final LottoNumbers lottoNumbers;
````
로 변경하면 아래처럼 필드에 직접 접근이 가능하고, 그러면 contains 함수를 없앨 수 있어요!
물론 LottoNumbers도 수정이 조금 필요하지만요ㅎ
```suggestion
        int matchCount = lottoNumbers.matchCount(winningLottoTicket.lottoNumbers);
```